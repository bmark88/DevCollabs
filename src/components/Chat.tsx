import React, { ReactNode } from "react"
import styled from "styled-components"
import { Message, MessageContainer, TimeStamp } from "../components/message"
// import App from "./hooks/App"

interface Props {
  users: any,
  messages: any,
  handleSubmit: any
}

const Div = styled.div`
border: solid;
border-color: #f0f0f0;
border-width: 2px;
min-width: 400px;
position: relative;
margin: 1em;
opacity: 80%;
// color: white;
// height: 800px;
// width: 20%;
// position: fixed;
// right: 0;
// float: right;
  
@media (max-width: 1000px) {
  display: none;
}
`
const UserList = styled.div`
  border-bottom: solid;
  background-color: #551A8B;
`

const SubmitButton = styled.form`
  bottom: 0;
  margin: 0;
  display: flex;
  height: 50px;
  position: absolute;
  width: 100%;
`;

const Input = styled.input`
  width: 100%;
`;

const Button = styled.button`
`;

const Chat = ({ users, messages, handleSubmit }: Props) => {
  return (
    <Div>
      <UserList>
        <h1>Online Members</h1>
        {users.map((user :string, index :number) => (
          <h3 key={index}>{user}</h3>
        ))}
      </UserList>
      <MessageContainer>
        {messages.map((msg: any) => (
          <>
            <b>{msg.user}: </b>{msg.message}
            <TimeStamp>{msg.date}</TimeStamp>
          </>
        ))}
      </MessageContainer>
      <SubmitButton onSubmit={handleSubmit}>
        <Input type="text" name="message" />
        <Button>Send</Button>
      </SubmitButton>

    </Div>
  )
}

export default Chat