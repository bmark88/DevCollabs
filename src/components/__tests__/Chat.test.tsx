import React from "react"
import renderer from "react-test-renderer"
// import { render } from "@types/testing-library__react"

import { render, cleanup } from "@testing-library/react"

import Chat from "../Chat"

const users = ["aliceLand1", "johnDoe2"]
const messages = ["hi", "hello"]

afterEach(cleanup)

describe("Chat", () => {
  xit("renders correctly", async () => {
    const tree = renderer
      .create(<Chat users={users} messages={messages} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
