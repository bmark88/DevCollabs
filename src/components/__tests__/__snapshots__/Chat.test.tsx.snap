// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat renders correctly 1`] = `
<div
  className="sc-AxhCb jGFQTj"
>
  <div
    className="sc-AxhUy hsSTiu"
  >
    <h4>
      Online Members
    </h4>
    <li
      className="sc-AxgMl lazBhf"
    >
      aliceLand1
    </li>
    <li
      className="sc-AxgMl lazBhf"
    >
      johnDoe2
    </li>
  </div>
  <div
    className="sc-AxiKw fZyRUb"
  >
    <span>
      <b>
        : 
      </b>
      <p
        className="sc-AxirZ jLJDQb"
      />
    </span>
    <span>
      <b>
        : 
      </b>
      <p
        className="sc-AxirZ jLJDQb"
      />
    </span>
  </div>
  <form
    className="sc-AxheI dzsoZX"
  >
    <div
      className="MuiFormControl-root MuiTextField-root WithStyles(ForwardRef(TextField))-root-1 MuiFormControl-fullWidth"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoComplete="off"
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          name="message"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Send a Message"
          required={false}
          type="text"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-40 MuiOutlinedInput-notchedOutline"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="PrivateNotchedOutline-legend-41"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Send
      </span>
    </button>
  </form>
</div>
`;
